homeassistant/components/qwikswitch/sensor.py

```diff
@@ -34,8 +34,8 @@ def __init__(self, sensor):
         sensor_type = sensor["type"]
 
         self._decode, self.unit = SENSORS[sensor_type]
-        if isinstance(self.unit, type):
-            self.unit = f"{sensor_type}:{self.channel}"
+        # this cannot happen because it only happens in bool and this should be redirected to binary_sensor
+        assert not isinstance(self.unit, type)
```

Comment:
will add a reason since if this happens, it can only be because of a bug in the code
there is also a problem that it is coupling the API from the library to the config in home assistant. probably makes sense to do create a map and then it cannot be broken if there are changes in the library

Reply:
ok. added the reason